draft-ietf-quic-transport.md

```diff
@@ -1640,89 +1643,90 @@ by a client in protected packets, because it is certain that the server is able
 to decipher the packet.
 
 
-## WINDOW_UPDATE Frame {#frame-window-update}
+## MAX_DATA Frame {#frame-max-data}
 
-The WINDOW_UPDATE frame (type=0x04) informs the peer of an increase in an
-endpoint's flow control receive window for either a single stream, or the entire
-connection as a whole.
+The MAX_DATA frame (type=0x04) is used in flow control to informs the peer of
+the maximum amount of data that can be sent on the connection as a whole.
 
 The frame is as follows:
 
 ~~~
  0                   1                   2                   3
  0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-|                        Stream ID (32)                         |
-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
 |                                                               |
-+                    Flow Control Offset (64)                   +
++                        Maximum Data (64)                      +
 |                                                               |
 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
 ~~~
 
-The fields in the WINDOW_UPDATE frame are as follows:
-
-Stream ID:
-
-: ID of the stream whose flow control windows is being updated, or 0 to specify
-  the connection-level flow control window.
-
-Flow Control Offset:
+The fields in the MAX_DATA frame are as follows:
 
-: A 64-bit unsigned integer indicating the flow control offset for the given
-  stream (for a stream ID other than 0) or the entire connection.
+Maximum Data:
 
-The flow control offset is expressed in units of octets for individual streams
-(for stream identifiers other than 0).
+: A 64-bit unsigned integer indicating the maximum amount of data that can be
+  sent on the entire connection, in units of 1024 octets.  That is, the updated
+  connection-level data limit is determined by multiplying the encoded value by
+  1024.
 
-The connection-level flow control offset is expressed in units of 1024 octets
-(for a stream identifier of 0).  That is, the connection-level flow control
-offset is determined by multiplying the encoded value by 1024.
+When counting data toward this limit, an endpoint accounts for the maximum
+offset of data that is sent or received on every stream.  Loss or reordering can
+mean that the maximum offset is greater than the total size of data received on
+a stream.  Receiving STREAM frames might not increase the maximum offset on a
```

Comment:
This sentence does not make sense: "Receiving STREAM frames might not increase the maximum offset on a stream if the maximum offset doesn't increase."